Use std::array for fluid property constants where applicable

cpgr opened the pull request:
To reduce temporary vector construction, the remaining vectors of
constants defined within several methods have been moved to the
header as well.

Closes #11373

cpgr committed:
Use std::array for fluid property constants where applicable

To reduce temporary vector construction, the remaining vectors of
constants defined within several methods have been moved to the
header as well.

Closes #11373

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/182403/) on 5c86d74 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11374/site)

This comment will be updated on new commits.

cpgr committed:
Increase dp in finite difference test of water viscosity

The derivative wrt pressure is very small, so increase the finite
difference parameter to provide a more accurate numerical
derivative, which enables the tolerance to be tightened.

Closes #11374
